sources/packages/backend/apps/api/src/route-controllers/student/student.external.controller.ts

```diff
@@ -42,17 +46,45 @@ export class StudentExternalController extends BaseController {
   @ApiNotFoundResponse({ description: "Student not found." })
   @HttpCode(HttpStatus.OK)
   async searchStudentDetails(
-    @Body() payload: ExternalSearchStudentAPIInDTO,
-  ): Promise<StudentDetailsAPIOutDTO> {
-    const student = await this.studentService.getStudentBySIN(payload.sin);
-    if (!student) {
+    @Body() payload: StudentSearchAPIInDTO,
+  ): Promise<StudentSearchResultAPIOutDTO> {
+    const studentPromise = this.studentInformationService.getStudentBySIN(
+      payload.sin,
+    );
+    const sfasIndividualPromise =
+      await this.studentInformationService.getSFASIndividualBySIN(payload.sin);
+    const [student, sfasIndividual] = await Promise.all([
+      studentPromise,
+      sfasIndividualPromise,
+    ]);
+    if (!student && !sfasIndividual) {
       throw new NotFoundException("Student not found.");
     }
+    // If the student is found in SIMS, return the student details.
+    // Otherwise, return the legacy student details.
+    const studentDetails = student
+      ? this.transformStudentDetails(student)
+      : this.transformLegacyStudentDetails(sfasIndividual);
```

Comment:
Is this a temporary solution or is it already the final approach to return student data?
I was assuming that the entire dataset of student + application data (a single query) would be executed (in parallel), targeting SIMS data and SFAS data. From the complete result of these two queries a "merge" would happen.
Is the idea that `sfasIndividualPromise` and `studentPromise` will be expanded?

Reply:
Initially, I thought of keeping student data fetch from DB separately, but as discussed student and application data will be retrieved in a single query in upcoming PR.